resources/agent.rb

```diff
@@ -35,32 +35,66 @@
                                    "backend-url": ['ws://127.0.0.1:8081'],
                                  }
 action :install do
-  packagecloud_repo new_resource.repo do
-    type value_for_platform_family(
-      %w(rhel fedora amazon) => 'rpm',
-      'default' => 'deb'
-    )
-  end
+  # Linux installation - source package
+  if node['platform'] != 'windows'
+    packagecloud_repo new_resource.repo do
+      type value_for_platform_family(
+        %w(rhel fedora amazon) => 'rpm',
+        'default' => 'deb'
+      )
+    end
 
-  package 'sensu-go-agent' do
-    if latest_version?(new_resource.version)
-      action :upgrade
-    else
-      version new_resource.version
-      action :install
+    # Installs/upgrades sensu-go-agent package
+    package 'sensu-go-agent' do
+      if latest_version?(new_resource.version)
+        action :upgrade
+      else
+        version new_resource.version
+        action :install
+      end
     end
-  end
 
-  # render template at /etc/sensu/agent.yml
-  file ::File.join(new_resource.config_home, 'agent.yml') do
-    content(stringify_keys(new_resource.config))
+    # render template at /etc/sensu/agent.yml for linux
+    file ::File.join(new_resource.config_home, 'agent.yml') do
+      content(stringify_keys(new_resource.config))
+    end
+
+    # Enable and start the sensu-agent service
+    service 'sensu-agent' do
+      if node['platform'] == 'ubuntu' && node['platform_version'].to_f == 14.04
+        provider Chef::Provider::Service::Init
+        action :start
+      else
+        action [:enable, :start]
+      end
+    end
   end
 
-  service 'sensu-agent' do
-    if node['platform'] == 'ubuntu' && node['platform_version'].to_f == 14.04
-      provider Chef::Provider::Service::Init
-      action :start
-    else
+  # Installs msi for Sensu-Go
+  if node['platform'] == 'windows'
+    windows_package 'sensu-go-agent' do
+      source 'https://s3-us-west-2.amazonaws.com/sensu.io/sensu-go/5.10.0/sensu-go-agent_5.10.0.4171_en-US.x64.msi'
+      installer_type :custom
+      version '5.10.0.4171'
+    end
+
+    # Adds install directory to path
+    windows_path 'c:\\Program Files\\sensu\\sensu-agent\\bin'
+
+    # render template at c:\Programdata\Sensu\config\agent.yml for windows
+    file ::File.join('c:/ProgramData/Sensu/config', 'agent.yml') do
+      content(stringify_keys(new_resource.config))
+    end
+
+    # Installs SensuAgent Service
+    powershell_script 'SensuAgent Service' do
+      code '.\\sensu-agent.exe service install'
+      cwd 'c:/Program Files/sensu/sensu-agent/bin'
```

Comment:
I see this being repeated, it should either be an internal variable or an exposed argument.

Reply:
Could you expand further? This is the method provided by Sensu.io to register SensuAgent as a service.

Reply:
```ruby
sensu_bindir = 'c:/Program Files/sensu/sensu-agent/bin'
...
windows_path sensu_bindir
...
powershell_script 'SensuAgent Service' do
      code '.\\sensu-agent.exe service install'
      cwd sensu_bindir
...
```